src/shared/components/lazyMobXTable/LazyMobXTableStore.tsx

```diff
@@ -0,0 +1,615 @@
+                        }}
+                    >
+                        {thContents}
+                        {this._columnToHeaderFilterIconModal(column)}
+                    </div>
+                );
+            }
+
+            let style: any = {};
+            if (column.align) {
+                style.textAlign = column.align;
+            }
+            if (column.width) {
+                style.width = column.width;
+            }
+
+            return (
+                <React.Fragment key={index}>
+                    <th
+                        ref={this.headerRefs[index]}
+                        className="multilineHeader"
+                        style={style}
+                    >
+                        {thContents}
+                    </th>
+                    {column.resizable && (
+                        <ColumnResizer
+                            className="multilineHeader columnResizer"
+                            minWidth={0}
+                        />
+                    )}
+                </React.Fragment>
+            );
+        });
+    }
+
+    @computed get visibleColumns(): Column<T>[] {
+        return this.columns.filter(column => this.isVisible(column));
+    }
+
+    @computed get colVisProp(): IColumnVisibilityDef[] {
+        const colVisProp: IColumnVisibilityDef[] = [];
+
+        this.columns.forEach((column: Column<T>) => {
+            colVisProp.push({
+                id: column.hasOwnProperty('id') ? column.id! : column.name,
+                name: column.name,
+                visible: this.isVisible(column),
+                togglable: column.hasOwnProperty('togglable')
+                    ? column.togglable
+                    : true,
+            });
+        });
+
+        return colVisProp;
+    }
+
+    @computed get paginationStatusText(): string {
+        let firstVisibleItemDisp;
+        let lastVisibleItemDisp;
+
+        if (this.rows.length === 0) {
+            firstVisibleItemDisp = 0;
+            lastVisibleItemDisp = 0;
+        } else {
+            firstVisibleItemDisp =
+                this.itemsPerPage === PAGINATION_SHOW_ALL
+                    ? 1
+                    : this.page * this.itemsPerPage + 1;
+
+            lastVisibleItemDisp =
+                this.itemsPerPage === PAGINATION_SHOW_ALL
+                    ? this.dataStore.totalElements
+                    : firstVisibleItemDisp + this.dataStore.itemsPerPage - 1;
+        }
+
+        let itemsLabel: string = this.itemsLabel;
+        if (itemsLabel.length) {
+            // we need to prepend the space here instead of within the actual return value
+            // to avoid unnecessary white-space at the end of the string
+            itemsLabel = ` ${itemsLabel}`;
+        }
+
+        return `Showing ${firstVisibleItemDisp}-${lastVisibleItemDisp} of ${this.dataStore.totalElements}${itemsLabel}`;
+    }
+
+    @computed get tds(): JSX.Element[][] {
```

Comment:
this probalby pre-existed this PR, but we should try to get JSX out of store.  it's just "wrong"

Reply:
yes, this code already existed